ccm_web/client/src/components/UserEnrollmentForm.tsx

```diff
@@ -0,0 +1,388 @@
+import React, { useState } from 'react'
+import { Backdrop, Button, CircularProgress, Grid, Link, makeStyles, Paper, Typography } from '@material-ui/core'
+
+import ErrorAlert from './ErrorAlert'
+import InlineErrorAlert from './InlineErrorAlert'
+import RoleSelect from './RoleSelect'
+import SectionSelectorWidget from './SectionSelectorWidget'
+import SuccessCard from './SuccessCard'
+import ValidatedFormField from './ValidatedFormField'
+import * as api from '../api'
+import usePromise from '../hooks/usePromise'
+import { CanvasCourseSectionWithCourseName, ClientEnrollmentType, getCanvasRole } from '../models/canvas'
+import { AddExternalUserEnrollment, AddNewExternalUserEnrollment } from '../models/enrollment'
+import { CanvasError } from '../utils/handleErrors'
+import { emailSchema, firstNameSchema, lastNameSchema, validateString, ValidationResult } from '../utils/validation'
+
+const useStyles = makeStyles((theme) => ({
+  spacing: {
+    marginBottom: theme.spacing(2)
+  },
+  buttonGroup: {
+    marginTop: theme.spacing(1)
+  },
+  alert: {
+    padding: theme.spacing(2)
+  },
+  container: {
+    position: 'relative',
+    zIndex: 0
+  },
+  backdrop: {
+    zIndex: theme.zIndex.drawer + 1,
+    color: '#FFF',
+    position: 'absolute',
+    textAlign: 'center'
+  }
+}))
+
+interface APIErrorWithContext {
+  error: Error
+  context: string
+}
+
+interface UserEnrollmentFormProps {
+  sections: CanvasCourseSectionWithCourseName[]
+  readonly rolesUserCanEnroll: ClientEnrollmentType[]
+  resetFeature: () => void
+  settingsURL: string
+}
+
+export default function UserEnrollmentForm (props: UserEnrollmentFormProps): JSX.Element {
+  const classes = useStyles()
+
+  const [selectedSection, setSelectedSection] = useState<CanvasCourseSectionWithCourseName | undefined>(undefined)
+
+  const [email, setEmail] = useState<string | undefined>(undefined)
+  const [emailValidationResult, setEmailValidationResult] = useState<ValidationResult | undefined>(undefined)
+  const [userExists, setUserExists] = useState<boolean | undefined>(undefined)
+  const [showIncompleteAlerts, setShowIncompleteAlerts] = useState<boolean>(false)
+
+  const [firstName, setFirstName] = useState<string | undefined>(undefined)
+  const [firstNameValidationResult, setFirstNameValidationResult] = useState<ValidationResult | undefined>(undefined)
+
+  const [lastName, setLastName] = useState<string | undefined>(undefined)
+  const [lastNameValidationResult, setLastNameValidationResult] = useState<ValidationResult | undefined>(undefined)
+  const [role, setRole] = useState<ClientEnrollmentType | undefined>(undefined)
+
+  const [success, setSuccess] = useState<true | undefined>(undefined)
+
+  const [doSearchForUser, isSearchForUserLoading, searchForUserError, clearSearchForUserError] = usePromise(
+    async (loginId: string): Promise<boolean> => {
```

Comment:
`loginId` is not being used in this block. should you remove it?

Reply:
or probably you will use it when you actually make a real API Call? in that case you can leave it

Reply:
Yeah, it will be used later.